lib/topology_lib_ip/library.py

```diff
@@ -320,6 +728,63 @@ def add_link_type_vlan(enode, portlbl, name, vlan_id, shell=None):
     enode.ports[name] = name
 
 
+def ip_addr(enode, shell=None):
+    """
+    Show the ip link table from the swns of a device
+
+    :param enode: Engine node to communicate with.
+    :type enode: topology.platforms.base.BaseNode
+    :rtype: dict
+    :return: A dictionary as returned by
+     :func: topology_lib_ip.parser._parse_ip_addr
+    """
+    d = None
+
+    # change into the appropriate context on device
+    cmd = 'ip netns exec swns ip addr'
+    response = enode(cmd, shell=shell)
+    d = _parse_ip_addr(response)
+    return d
+
+
+def ip_link(enode, shell=None):
+    """
+    Show the ip link table from the swns of a device
+
+    :param enode: Engine node to communicate with.
+    :type enode: topology.platforms.base.BaseNode
+    :rtype: dict
+    :return: A dictionary as returned by
+     :func: topology_lib_ip.parser._parse_ip_link
+    """
+    d = None
+
+    # change into the appropriate context on device
+    cmd = 'ip netns exec swns ip link'
```

Comment:
This code only works with a container with openswitch and the library should be abstract to reuse it across containers with other images